File: web/resources/js/Mensagem.js
var Mensagem;

Mensagem = (function() {

    /**
     * @param String tipo
     * @param String classe
     * @param Object dados
     * @param String erro
     */
    function Mensagem(tipo, classe, dados, erro) {
        if (classe == null && dados == null && erro == null) {
            var obj = (typeof tipo == "string" ? JSON.parse(tipo) : tipo);
            tipo = obj.tipo;
            classe = obj.classe;
            dados = obj.dados;
            erro = obj.erro;
        }
        this.tipo = tipo;
        this.classe = classe;
        this.dados = dados;
        this.erro = erro;
    }

    return Mensagem;
})();

File: src/java/websocket/entidade/mensagem/Mensagem.java
package websocket.entidade.mensagem;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

/**
 *
 * @author rubensmarcon
 */
public class Mensagem {
       
    protected String tipo;
    //private String classe;
    protected Object dados;
    protected String erro;

    public Mensagem(String tipo, Object dados, String erro) {
        this.tipo = tipo;
        this.dados = dados;
        this.erro = erro;
    }
    
    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public Object getDados() {
        return dados;
    }

    public void setDados(Object dados) {
        this.dados = dados;
    }

    public String getErro() {
        return erro;
    }

    public void setErro(String erro) {
        this.erro = erro;
    }

    @Override
    public String toString() {
        return "Mensagem{" + "tipo=" + tipo + ", dados=" + dados + ", erro=" + erro + '}';
    }

    
}
